template/Berksfile.erb

```diff
@@ -8,3 +8,5 @@ source '<%= s %>'
 <% cookbook.depends.each do |name, cookbook| -%>
 cookbook '<%= name %>', '<%= cookbook.fetch(:version, '>= 0.0.0') %>', <%= cookbook.to_hash %>
 <% end -%>
+
+solver :<%= berkshelf.solver %>
```

Comment:
Do you know if `ERB` does a `to_s` on symbols before exposing them to the template? I genuinely don't know, but if it doesn't your output will be `solver ::gecode` (or whatever).